deepinv/sampling/sampling.py

```diff
@@ -0,0 +1,366 @@
+import sys
+from collections import deque
+from typing import Union, Dict, Callable, List, Tuple
+
+import torch
+from tqdm import tqdm
+
+from deepinv.models import Reconstructor
+from deepinv.optim.data_fidelity import DataFidelity
+from deepinv.optim.prior import Prior
+from deepinv.optim.utils import check_conv
+from deepinv.physics import Physics, LinearPhysics
+from deepinv.sampling.sampling_iterators import *
+from deepinv.sampling.utils import Welford
+
+
+class BaseSampling(Reconstructor):
+    r"""
+    Base class for Monte Carlo sampling.
+
+    This class can be used to create new Monte Carlo samplers by implementing the sampling kernel through :class:`deepinv.sampling.SamplingIterator`:
+
+    ::
+
+        # define your sampler (possibly a Markov kernel which depends on the previous sample)
+        class MyIterator(SamplingIterator):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x, y, physics, data_fidelity, prior, params_algo):
```

Comment:
x or X?

Reply:
Worth double checking and making clear everywhere when tensors are expected and when dicts are expected